services/cd-service/pkg/repository/repository_test.go

```diff
@@ -1511,3 +1515,71 @@ func TestProcessQueueOnce(t *testing.T) {
 		})
 	}
 }
+
+func TestGitPushDoesntGetStuck(t *testing.T) {
+	tcs := []struct {
+		Name string
+	}{
+		{
+			Name: "it doesnt get stuck",
+		},
+	}
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			// create a remote
+			dir := t.TempDir()
+			remoteDir := path.Join(dir, "remote")
+			localDir := path.Join(dir, "local")
+			cmd := exec.Command("git", "init", "--bare", remoteDir)
+			cmd.Run()
+			ts := testssh.New(remoteDir)
+			defer ts.Close()
+			repo, err := New(
+				ctx,
+				RepositoryConfig{
+					URL: ts.Url,
+					Certificates: Certificates{
+						KnownHostsFile: ts.KnownHosts,
+					},
+					Credentials: Credentials{
+						SshKey: ts.ClientKey,
+					},
+					Path:           localDir,
+					NetworkTimeout: time.Second,
+				},
+			)
+			if err != nil {
+				t.Errorf("expected no error, got %q ( %#v )", err, err)
+			}
+			err = repo.Apply(testutil.MakeTestContext(),
+				&CreateEnvironment{Environment: "dev"},
+			)
+			if err != nil {
+				t.Errorf("expected no error, got %q ( %#v )", err, err)
+			}
+			// This will prevent the next push from working
+			ts.DelayExecs(15 * time.Second)
+			err = repo.Apply(testutil.MakeTestContext(),
+				&CreateEnvironment{Environment: "stg"},
+			)
+			if err == nil {
+				t.Errorf("expected an error, but didn't get one")
+			}
+			if status.Code(err) != codes.Canceled {
+				t.Errorf("expected status code cancelled, but got %q", status.Code(err))
+			}
+			// This will prevent the next push from working
```

Comment:
Will it be prevented? Technically there is no delay or am I missing something?

Reply:
No, you are right. The wording must be the other way round.